lapack/gonum/dtgsy2.go

```diff
@@ -0,0 +1,674 @@
+						bi.Dger(m-ie-1, nb, -1, d[is*ldd+ie+1:], 1, rhs[2:], 1, c[(ie+1)*ldc+js:], ldc)
+					}
+				} else if mb == 2 && nb == 1 {
+					// Build a 4×4 system Zᵀ * x = RHS.
+					z[0] = a[is*lda+is]
+					z[ldz] = a[is*lda+isp1]
+					z[2*ldz] = -b[js*ldb+js]
+					z[3*ldz] = 0
+
+					z[1] = a[isp1*lda+is]
+					z[ldz+1] = a[isp1*lda+isp1]
+					z[2*ldz+1] = 0
+					z[3*ldz+1] = -b[js*ldb+js]
+
+					z[2] = d[is*ldd+is]
+					z[ldz+2] = d[is*ldd+isp1]
+					z[2*ldz+2] = -e[js*lde+js]
+					z[3*ldz+2] = 0
+
+					z[3] = 0
+					z[ldz+3] = d[isp1*ldd+isp1]
+					z[2*ldz+3] = 0
+					z[3*ldz+3] = -e[js*lde+js]
+
+					// Set up right hand side(s).
+					rhs[0] = c[is*ldc+js]
+					rhs[1] = c[isp1*ldc+js]
+					rhs[2] = f[is*ldf+js]
+					rhs[3] = f[isp1*ldf+js]
+
+					// Solve Zᵀ * x = RHS
+					k = impl.Dgetc2(zdim, z, ldz, ipiv[:zdim], jpiv[:zdim])
+					if k >= 0 {
+						info = k
+					}
+					scaloc = impl.Dgesc2(zdim, z, ldz, rhs, ipiv[:zdim], jpiv[:zdim])
+					if scaloc != 1 {
+						for k = 0; k < n; k++ {
+							bi.Dscal(m, scaloc, c[k:], ldc)
+							bi.Dscal(m, scaloc, f[k:], ldf)
+						}
+						scale *= scaloc
+					}
+
+					// Unpack solution vectors
+					c[i*ldc+js] = rhs[0]
+					c[isp1*ldc+js] = rhs[1]
+					f[is*ldf+js] = rhs[2]
+					f[isp1*ldf+js] = rhs[3]
+
+					// Substitute R(i, j) and L(i, j) into remaining equation.
+					if j > p+2 {
+						bi.Dger(mb, js, 1, rhs, 1, b[js:], ldb, f[is*ldf:], ldf)
+						bi.Dger(mb, js, 1, rhs[2:], 1, e[js:], lde, f[is*ldf:], ldf)
+					}
+					if i < p {
+						bi.Dgemv(blas.Trans, mb, m-ie-1, -1, a[is*lda+ie+1:], lda, rhs, 1,
+							1, c[(ie+1)*ldc+js:], ldc)
+						bi.Dgemv(blas.Trans, mb, m-ie-1, -1, d[is*ldd+ie+1:], ldd, rhs[2:], 1,
+							1, c[(ie+1)*ldc+js:], ldc)
+					}
+
+				} else if mb == 2 && nb == 2 {
+					// Build 8×8 system Zᵀ * x = RHS
+					impl.Dlaset(blas.All, ldz, ldz, 0, 0, z, ldz)
+
+					z[0] = a[is*lda+is]
+					z[ldz] = a[is*lda+isp1]
+					z[4*ldz] = -b[js*ldb+js]
+					z[6*ldz] = -b[jsp1*ldb+js]
+
+					z[1] = a[isp1*lda+is]
+					z[ldz+1] = a[isp1*lda+isp1]
+					z[5*ldz+1] = -b[js*ldb+js]
+					z[7*ldz+1] = -b[jsp1*ldb+js]
+
+					z[2*ldz+2] = a[is*lda+is]
+					z[3*ldz+2] = a[is*lda+isp1]
+					z[4*ldz+2] = -b[js*ldb+jsp1]
+					z[6*ldz+2] = -b[jsp1*ldb+jsp1]
+
+					z[2*ldz+3] = a[isp1*lda+is]
+					z[3*ldz+3] = a[isp1*lda+isp1]
+					z[5*ldz+3] = -b[js*ldb+jsp1]
+					z[7*ldz+3] = -b[jsp1*ldb+jsp1]
+
+					z[4] = d[is*ldd+is]
+					z[1*ldz+4] = d[is*ldd+isp1]
+					z[4*ldz+4] = -e[js*lde+js]
+
+					z[ldz+5] = d[isp1*ldd+isp1]
+					z[5*ldz+5] = -e[js*lde+js]
+
+					z[2*ldz+6] = d[is*ldd+is]
+					z[3*ldz+6] = d[is*ldd+isp1]
+					z[4*ldz+6] = -e[js*lde+jsp1]
+					z[6*ldz+6] = -e[jsp1*lde+jsp1]
+
+					z[3*ldz+7] = d[isp1*ldd+isp1]
+					z[5*ldz+7] = -e[js*lde+jsp1]
+					z[7*ldz+7] = -e[jsp1*lde+jsp1]
+
+					// Set up right hand side(s).
+					k = 0
+					ii := nb * mb
+					for jj := 0; jj < nb-1; jj++ {
```

Comment:
Looks like a bug to me. reference starts at jj at 0 and iterates to nb-1, should be `<=`